test/test-esm-named-exports.tap.mjs

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+import tap from 'tap'
+var test = tap.test
+import { wrap, unwrap, massWrap, massUnwrap } from '../index.js'
+
+var outsider = 0
+function counter() { return ++outsider }
+function anticounter() { return --outsider }
+
+var generator = {
+  inc: counter
+}
+Object.defineProperty(generator, 'dec', {
+  value: anticounter,
+  writable: true,
+  configurable: true,
+  enumerable: false
+})
+
+test('should wrap safely with named export in esm', function (t) {
```

Comment:
This is lifted from the `wrap.tap.js` test, with the main difference being the use of the named import. I wasn't sure whether to add tests for each method or if this is sufficient.